test/infrastructure/docker/exp/controllers/dockermachinepool_controller.go

```diff
@@ -179,8 +179,8 @@ func (r *DockerMachinePoolReconciler) reconcileNormal(ctx context.Context, clust
 		return ctrl.Result{}, errors.Wrap(err, "failed to build new node pool")
 	}
 
-	// if we don't have enough nodes matching spec, build them
-	if err := pool.ReconcileMachines(ctx); err != nil {
+	// Reconcile machines and updates Status.Instances
+	if err := pool.ReconcileMachines(ctx, log); err != nil {
```

Comment:
Given that this is v1alpha4, we should start using the logger from the context instead of passing it as a parameter

Reply:
this is a little bit out of the scope of this PR, but given that it is a nice cleanup, Done!